test/modules/packages/collections/CollectionCounter.chpl

```diff
@@ -51,9 +51,10 @@ var barrier = new Barrier(here.maxTaskPar * numLocales);
 coforall loc in Locales do on loc {
   var perLocaleActual : atomic int;
   const _c = c;
+  var (hasElem, elt) : (bool, int);
   coforall tid in 0..#here.maxTaskPar {
     barrier.barrier();
-    var (hasElem, elt) : (bool, int) = (true, 0);
+    (hasElem, elt) = (true, 0);
     var perTaskActual : int;
     while hasElem {

```

Comment:
@ronawho 

```
A while loop with a constant condition
```

Reply:
Hmm, this check occurs long before LICM is actually run, so I think it's unrelated to that. Does this cause any failures beyond getting the warning itself?

https://github.com/chapel-lang/chapel/issues/6532 has some related info

Reply:
@ronawho 

No errors, get a good old `SUCCESS`, thanks for the fix!

@mppf 

Seeing as you merged a patch for infinite loop checking, was wondering if you could take a look on this when you have the time.

Reply:
In fact, if I got some information on where to go to approach this myself, I can potentially attempt it myself (such as, where these checks occur, potentially 'why' it occurs, a decent tour around the Chapel compiler, etc.)

Reply:
For this PR I think we should move the declaration back down to avoid the `A while loop with a constant condition` warning. That warning code either needs to be improved (or IMO removed), but I think we should that in a separate PR.

Reply:
And FYI: the check occurs in `compiler/AST/WhileStmt.cpp::checkConstWhileLoop()` as part of `WhileStmt::checkConstLoops()`, which is ultimately called during `functionResolution()`

The warning is (clearly) inaccurate, and at least I don't think it's very valuable anyways

Reply:
So, tracing it, it looks like the issue is [this](https://github.com/chapel-lang/chapel/blob/master/compiler/AST/WhileStmt.cpp#L313-L315), that it incorrectly sees it as a constant expression. It seems the issue is [here](https://github.com/chapel-lang/chapel/blob/master/compiler/AST/WhileStmt.cpp#L320)? It only goes one parent up, when it seems to need to check all parents.

Reply:
In particular, [this](https://github.com/chapel-lang/chapel/blob/master/compiler/AST/WhileStmt.cpp#L320) line may stop it too early, if `parent == this`

Reply:
Actually, nevermind, ignore that, issue would be that it sees `condSym` as `true` to begin with. I'll just investigate silently until I actually arrive at the conclusion.

Reply:
@ronawho 

Sorry to bother you again, but is it possible for `GotoStmt::gotoTag` to be this value: `42262848`? If the answer is no, its possible that it got mangled or something, because if you look at the type, its an `enum GotoTag`, defined [here](https://github.com/chapel-lang/chapel/blob/e67f28cfd0a81094e05830c9cc6fb55da97a9f38/compiler/include/stmt.h#L191-L201). As can be seen, an `enum` shouldn't have an undefined value unless it wasn't initialized correctly, leading to undefined behavior.  I got the value by using gdb. If this is due to it simply being uninitialized its an easy fix but can lead to some major complications down the road if anything else wanted to use any other other uninitialized values. In any event, this undefined value causes it to fail [this](https://github.com/chapel-lang/chapel/blob/master/compiler/AST/WhileStmt.cpp#L356) check.

Actually, after typing this, it seems that it may be a pointer, and thats the issue. Casting it in gdb to another `GotoStmt *` shows that it directs to something with `GOTO_NORMAL` tag. Perhaps the issue is that its too specific with what it expects, or is there a reason that it only checks for `return` and `break`?

Reply:
Regarding the compiler changes - might be time to move the discussion to an issue / another PR. I personally wouldn't cry if we entirely removed those checks for infinite loops though.

Regarding GotoTag having a weird value - usually I try do debug that sort of thing by running the compiler in valgrind.